### ISSUE TITLE
How to opt out from any collected data? This surviellance is not acceptable.

### ISSUE THREAD

**Issue by barshag** [joaomdmoura/crewAI#524]

**Comment by theCyberTech** [comment]
Read the docs - https://docs.crewai.com/telemetry/Telemetry/#data-collected-includes

**Comment by b1ochicken** [comment]
I have to agree with @barshag, there needs to be a way to totally disable it.  

Perhaps instead of hard wiring AgentOps in a telemetry provider an interface could be swapped in.  A default  DevNullTelemetryProvider should be the default,  By all means provide an optional AgentOpsTelemetryProvider but don't force it.

CrewAI now depends on AgentOps and you can't opt out of AgentOps TOS.   This will cause complications within corporate environments for sure.  https://github.com/AgentOps-AI/agentops/blob/main/TOS.md 

Can this be rolled back until a better solution is provided?

**Comment by theCyberTech** [comment]
Crew Telemetry is disabled by default - you have to opt-in

**Comment by b1ochicken** [comment]
You do have to include AgentOps now as the crew.py, agents.py, tool_usage.py, now all "import agentops" otherwise you'll get a dependency error.

**Comment by joaomdmoura** [comment]
We are rolling back agentsops integration for now, it is on a RC so it's not out on a proper version only for people manually installing that specific version.

The crew telemetry is actually enabled but anonymous by default, maybe - you have to opt-in to make it non-anonymous. We are also putting out our own TOS.

**Comment by piratos** [comment]
I think  you can get Crewai, python ans OS versions stats from pypi directly

**Comment by barshag** [comment]
> The crew telemetry is actually enabled but anonymous by default, maybe - you have to opt-in to make it non-anonymous. We are also putting out our own TOS.

**** Why?****

- Even Microsoft  allow you to opt-out from annonymous by default (on windows)!
- This was meant to be community effort project and now it's all about survelliance to gain benefits from usages.
- This is wrong practical of any Open-source project ethic. You can see it in NextJS or any other big project.

If one is willing to give you data fine, but why make it mandatory for all of us that our privacy is so important for us?

**Comment by barshag** [comment]
But it's still relevnt
